### ISSUE TITLE
Compilation error due to missing import in generated code when using <lambdaConverter/> on a routine

### ISSUE THREAD

**Issue by KaRkY** [jOOQ/jOOQ#14524]
### Actual behavior

Generated class that extends AbstractRoutine is missing import for "org.jooq.Converter".

### Steps to reproduce the problem

```
<forcedType>
  <userType>example.SlugBase</userType>
   <lambdaConverter>
    <from>SlugBase::new</from>
    <to>SlugBase::value</to>
  </lambdaConverter>
  <includeExpression>.*\.SLUG_BASE</includeExpression>
</forcedType>
```

### jOOQ Version

3.17.7

### Database product and version

PostgreSQL

### Java Version

openjdk 19

### OS Version

Windows 10

### JDBC driver name and version (include name if unofficial driver)

postgresql:42.5.1

**Comment by KaRkY** [comment]
For reference my current workaround is
```
<forcedType>
  <userType>example.SlugBase</userType>
  <converter>org.jooq.Converter.ofNullable(String.class, SlugBase.class, SlugBase::new, SlugBase::value)</converter>
  <includeExpression>.*\.SLUG_BASE</includeExpression>
</forcedType>
```

**Comment by lukaseder** [comment]
Thanks for your report. Interesting, might be routine related. Our integration tests only cover `<lambdaConverter/>` on tables, where the import is present. Will try to reproduce.

**Comment by lukaseder** [comment]
Yep, reproduces simply. Will fix right away.

**Comment by lukaseder** [comment]
To make things even, the import is generated in the `Routines.java` file, where it isn't needed.

**Comment by lukaseder** [comment]
Might be a caching effect. The `RoutineDefinition` is created only once from the first caller. In this case, the `Routines.java` class is generated first, then the routines themselves. This leads to the wrong `JavaTypeResolver` instance being cached with the `RoutineDefinition`, that of `Routines.java`, not the actual routine.

There's probably a much bigger underlying caching problem, design-wise, which might cause other subtle issues, elsewhere. But for this particular problem, a quick fix is to simply reverse the order of generation:

1. Routine bodies
2. `Routines.java`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.8 (https://github.com/jOOQ/jOOQ/issues/14525), 3.16.14 (https://github.com/jOOQ/jOOQ/issues/14526).

Thanks again for your report!
